design-patterns/creational-patterns/src/objectPool/ReusablePool.java

```diff
@@ -0,0 +1,49 @@
+package objectPool;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReusablePool {
+	private List<Reusable> availableReusables;
+	private List<Reusable> usedReusables;
```

Comment:
takenReusables or lockedReusables is better name I believe. usedReusables says that they are used by someone sometime, not that someone using them at the moment